File: Runde.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Quizly
{
    public class Runde
    {
        public int RundeId { get; set; }
        public int RundenNummer { get; set; }

        public Kategorie Kategorie { get; set; }
        public DateTime StartZeit { get; set; }
        public DateTime? EndZeit { get; set; }

        public List<Frage> Fragen { get; set; }

        public void Beenden()
        {
            EndZeit = DateTime.UtcNow;
        }
    }
}

File: Android/app/src/main/java/com/quizly/android/data/model/Runde.kt
package com.quizly.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Ported from Runde.cs / runde.py. */
@Serializable
data class Runde(
    @SerialName("runde_id")
    val rundeId: Int,
    @SerialName("runden_nummer")
    val rundenNummer: Int,
    @SerialName("kategorie_id")
    val kategorieId: Int? = null,
    @SerialName("kategorie_name")
    val kategorieName: String? = null,
    @SerialName("start_zeit")
    val startZeit: String? = null,
    @SerialName("end_zeit")
    val endZeit: String? = null,
    val fragen: List<Frage> = emptyList()
)
